### ISSUE TITLE
Scheduler Pending Timeout waiting for VM resources with OneFlow Service Templates

### ISSUE THREAD

**Issue by KenM603** [OpenNebula/one#6832]
**Description**
Currently a new VM create request sent to the One Scheduler service will stay in PENDING state forever waiting on available resources (Memory, CPU, storage).   The request is to provide a method to fail the request after a configurable time period.

**Use case**
At the Scheduler we can think of this as 1 VM.  However, for our use-case this becomes more important with OneService that has many VMs tied together with a "role" relationship.    For the service to work properly all initial VMs are required.   We have automation creating many new independent Service environments on Demand via Open Nebula OneFlow APIs.

At times we run out of KVM Host resources.    Since currently the Scheduler will stay in VM Pending state forever, then the status of  OneFlow Template create will also stay in "Deploying" forever.   This becomes difficult for an automated infra.   Ideally this will fail at some point and return an Error code on OneFlow.

**Interface Changes**

- [ ] Minimum change requested:
/etc/one/sched.conf  should have user configurable PENDING TIMEOUT value.

- [ ] Nice to have aditional change requested:
Add a VM Templates new configurable to pass PENDING TIMEOUT to Scheduler.     This can override the configured Scheduler default value.   This can be useful for larger VMs that might take longer than default PENDING TIMEOUT.

**Additional Context**

Please see this Forum post for additional comments:
   https://forum.opennebula.io/t/service-template-deployment-pending-forever-waiting-on-host-resources/13362

![image](https://github.com/user-attachments/assets/01360613-1f5f-43aa-9fda-a9235d07fc90)

## Progress Status
- [ ] Code committed
- [ ] Testing - QA
- [ ] Documentation (Release notes - resolved issues, compatibility, known issues)
